matrix/SquareMatrix.hpp

```diff
@@ -133,7 +149,7 @@ bool inv(const SquareMatrix<Type, M> & A, SquareMatrix<Type, M> & inv)
     for (size_t n = 0; n < M; n++) {
 
         // if diagonal is zero, swap with row below
-        if (fabs(static_cast<float>(U(n, n))) < 1e-8f) {
+        if (fabs(static_cast<float>(U(n, n))) < __FLT_EPSILON__) {
```

Comment:
```suggestion
        if (fabs(static_cast<float>(U(n, n))) < FLT_EPSILON) {
```

Reply:
Done. I add to define it so it could work.